.github/workflows/docker-ci.yml

```diff
@@ -0,0 +1,628 @@
+            platform="linux/arm64"
+            platform_tag="arm"            
+          elif grep -q -E '^[^#]*LABEL\s+org\.opencontainers\.image\.arch\s*=\s*(x86|amd64|x86_64)' "$file"; then
+            platform="linux/amd64"
+            platform_tag="x86"
+          else
+            echo "No known architecture label found in $file. Exiting."
+            exit 1
+          fi
+          echo "Platform: $platform"
+          echo "Platform tag: $platform_tag"
+          echo "dockerfile_name=$dockerfile_name" >> $GITHUB_OUTPUT
+          echo "platform=$platform" >> $GITHUB_OUTPUT
+          echo "platform_tag=$platform_tag" >> $GITHUB_OUTPUT
+          echo "directory=$dir" >> $GITHUB_OUTPUT
+          exit 0 
+
+      - name: Set current date
+        if: steps.set_proceed_flag.outputs.proceed_valid == 'true'
+        id: date
+        run: |
+          date_tag=$(date +'%m-%d')
+          echo "Date tag: $date_tag"
+          echo "date=$date_tag" >> $GITHUB_OUTPUT
+      
+      - name: Determine runner label
+        if: steps.set_proceed_flag.outputs.proceed_valid == 'true'
+        id: determine_runner
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{secrets.PAT_TOKEN}}
+          script: |
+            let runner_label = 'X64';
+            const platform = '${{ steps.parse_path.outputs.platform }}';
+            if (platform === 'linux/arm64') {
+              const runners = await github.paginate(
+                github.rest.actions.listSelfHostedRunnersForRepo,
+                {
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                }
+              );
+
+              console.log(`Total runners found: ${runners.length}`);
+
+              runners.forEach(runner => {
+              const labels = runner.labels.map(label => label.name);
+              console.log(`Runner ID: ${runner.id}, Name: ${runner.name}, Status: ${runner.status}, Labels: ${labels.join(', ')}`);
+              });
+
+              let found = false;
+              for (const runner of runners) {
+                if (runner.status === 'online') {
+                  const labels = runner.labels.map(label => label.name);
+                  if (labels.includes('ARM64')) {
+                    runner_label = 'ARM64';
+                    console.log(`Found online runner with label "ARM64": Runner ID ${runner.id}, Name ${runner.name}`);
+                    found = true;
+                    break;
+                  }
+                }
+              }
+              if (found) {
+                console.log('Found online runner with label "ARM64"');
+              } else {
+                console.log('No online runner with label "ARM64" found, using default runner');
+              }
+            } else {
+              console.log('Platform is not linux/arm64, using default X64 runner');
```

Comment:
I might be missing something but is there a reason we don't check if the `X64` runner is online?